### ISSUE TITLE
Azure Cloud Shell intermittently doesn't work for some users

### ISSUE THREAD

**Issue by PankajBhojwani** [microsoft/terminal#16169]
### Windows Terminal version

1.17

### Steps to reproduce

1. Open up an Azure Cloud Shell tab
2. Go through the login process

### Expected Behavior

Logs in successfully

### Actual Behavior

<img width="1280" alt="image" src="https://github.com/microsoft/terminal/assets/26824113/d69e926b-e437-484f-88e3-dc91fb6ab6d4">

**Comment by PankajBhojwani** [comment]
(screenshot above is from an Azure Support employee messaging me on Teams)

This is the second of this kind of report we have received recently so there's likely something here. I have collected traces of the login process above and the current suspicion is that there was an API change on the Azure side and we aren't conforming to the new API. We have set up a meeting with the Cloud Shell team, will update here once that's done.
